### DISCUSSION TITLE
Help, trying to apply a "baseStyle" to a chakra component using "extendTheme"

### DISCUSSION THREAD

**Discussion by ghost** [chakra-ui/chakra-ui#5339] [Q&A]
Hi, i need help because i have 2 chakra components: 
* Link
* Circle

The problem is that when i apply the baseStyle to the Link component, it works, but it doesn't when is with Circle.

The Circle component is wrapped in the Link component

* I'm using VITEJS with ReactTypescript.
* I have the following folders structure:
 
![image](https://user-images.githubusercontent.com/70576040/148408909-718bcdf6-dc91-4870-a095-c9b4e35e9f53.png)

## Code:

* Index File inside **src/theme**

```ts
import { extendTheme } from '@chakra-ui/react';

import styles from './styles';
import components from './components';

export default extendTheme({...styles, components });
```
* Index File inside **src/theme/components**

```ts
import { Link } from './Link'; 
import { Circle } from './Circle'; 


export default {
  Link,
  Circle
}
```
* Link File inside **src/theme/components**

```ts
import { ComponentSingleStyleConfig } from '@chakra-ui/react';

export const Link = {
  baseStyle:({colorMode})=>({
    backgroundColor:'#8B3B8F',
    color: colorMode === 'light' ? '#06008B' : '#ccc',
    transition:'background-color 1s linear',
    _hover:{
      backgroundColor: 'red',
    }
  }),
} as ComponentSingleStyleConfig;
```
* Circle File inside **src/theme/components**
```ts
import { ComponentSingleStyleConfig } from '@chakra-ui/react';

export const Circle = {
  baseStyle:({colorMode})=>({
    backgroundColor:'#7ff5a1',
    color: colorMode === 'light' ? '#06008B' : '#ccc',
    transition:'background-color 1s linear',
    _hover:{
      backgroundColor: 'red',
    }
  }),
} as ComponentSingleStyleConfig;
```

* Component "Main" where i'm using this chakra components.
```jsx
import { Box, Circle, Square, Icon, Heading, AbsoluteCenter, Flex, Grid, GridItem ,Link } from '@chakra-ui/react'
import { LinksToRedirect } from '@utils/DefinedValues/pages/Main_LinksToRedirect';


export default function Main(){
  return(
  <Box minW='100vw' minH='100vh'>
    <AbsoluteCenter w='fit-content'>
      <Flex flexDir={'column'} rowGap='5rem' textAlign={'center'}>
      <Heading>
        You don't have an <Link>Account</Link> yet?
      </Heading>
      <Grid  gridTemplateColumns={'1fr 1fr'}>

          {
            LinksToRedirect.map(
              ({to, text, type, icon}: typeof LinksToRedirect[0], idx:number)=>
              (
                <GridItem key={`${type}_${idx}`} p='1em' rowGap='1em'>
                <Link href={to}>
                    <Circle>
                      <Icon as={icon} />
                      {text}
                    </Circle>
                </Link>
              </GridItem>            
            ))
          }
      </Grid>
      </Flex>
    </AbsoluteCenter>
  </Box>
)
}
```

* My Page, Notice That **Account**, is applying the **baseStyle** meanwhile The **SignIn** **SignUp** doesn't.
![image](https://user-images.githubusercontent.com/70576040/148411442-faa76f4c-670c-4920-b82b-dfd00df6de77.png)

**Comment by primos63** [comment]
Well there's bad news and good news.

**The Bad News**
Not all components utilize the theme. `Circle` does not so it's working exactly as it was designed.

Let's look at the [source code](https://github.com/chakra-ui/chakra-ui/blob/main/packages/layout/src/box.tsx) for `Box` which also contains the  source for `Square` and `Circle`:
```jsx
export const Square = forwardRef<SquareProps, "div">((props, ref) => {
  const { size, centerContent = true, ...rest } = props

  const styles: SystemStyleObject = centerContent
    ? { display: "flex", alignItems: "center", justifyContent: "center" }
    : {}

  return (
    <Box
      ref={ref}
      boxSize={size}
      __css={{
        ...styles,
        flexShrink: 0,
        flexGrow: 0,
      }}
      {...rest}
    />
  )
})

if (__DEV__) {
  Square.displayName = "Square"
}

export const Circle = forwardRef<SquareProps, "div">((props, ref) => {
  const { size, ...rest } = props
  return <Square size={size} ref={ref} borderRadius="9999px" {...rest} />
})

if (__DEV__) {
  Circle.displayName = "Circle"
}
```

`Circle` is a `Square` with a border radius of 9999px which makes it round. `Square` has styles, but they don't come from the theme.
```javascript
const styles: SystemStyleObject = centerContent
    ? { display: "flex", alignItems: "center", justifyContent: "center" }
    : {}
```
They are hard-coded. So no matter what you put in the theme for `Square` or `Circle` it's not going to matter.

**The Good News**
If you want to get styles from the theme, then you can create your own `Circle` component which requests the theme style and returns Chakra's `Square` component with the styles placed in the `__css` prop and of course setting the `borderRadius` prop to 9999px. You can follow the example in the [documentation](https://chakra-ui.com/docs/theming/component-style#return-value) for getting the styles from the theme for a component.

**Reply by ghost** [reply]
Thanks, it was perfect to understand why it wasn't workimg
